homeassistant/components/solarlog/__init__.py

```diff
@@ -57,12 +57,14 @@ async def async_migrate_entry(
                         entity.entity_id, new_unique_id=new_uid
                     )
 
+        if config_entry.minor_version < 3:
             # migrate config_entry
             new = {**config_entry.data}
-            new["extended_data"] = False
+            new.pop("extended_data", new)
```

Comment:
We don't have to pop right?

Reply:
I thought if this property is not needed anymore, we should not keep it in the records. Or does HA take care of this on its own?

Reply:
It doesn't, but it's a minor version migration, so if users revert back to a previous version of core, it should still work. By popping extended data, we break that

Reply:
Ok, makes sense. Is there a process how to track such deprecated properties and when they can be deleted? Or should I leave this forever in the config entry?

Reply:
Or should I pop it and add a routine for downgrading to minor version 2?